operator/pkg/resources/statefulset.go

```diff
@@ -111,7 +112,10 @@ type StatefulSetResource struct {
 	// annotation to ensure the pods get restarted when configuration changes
 	// this has to be retrieved lazily to achieve the correct order of resources
 	// being applied
-	nodeConfigMapHashGetter  func(context.Context) (string, error)
+	nodeConfigMapHashGetter func(context.Context) (string, error)
+	// The configuration we've installed. It's injected so that we can extract any additional
+	// environment variables the configuration requires to expand the bootstrap template.
+	configurationGetter      func(ctx context.Context) (*configuration.GlobalConfiguration, error)
```

Comment:
I'm not sure what contexts this is used in but could this just be `[]corev1.EnvVar` rather than a closure to limit how much of the configuration "leaks" into the STS generation?